core/app/models/spree/payment_method.rb

```diff
@@ -24,9 +24,9 @@ class UnsupportedPaymentMethod < StandardError; end
 
     validates :name, :type, presence: true
 
-    has_many :payments, class_name: "Spree::Payment", inverse_of: :payment_method
-    has_many :credit_cards, class_name: "Spree::CreditCard"
-    has_many :store_payment_methods, inverse_of: :payment_method
+    has_many :payments, class_name: "Spree::Payment", inverse_of: :payment_method, dependent: :restrict_with_exception
```

Comment:
Here we could also use `restrict_with_error` instead and display a nice error to the user, right?